desc/compute/_stability.py

```diff
@@ -0,0 +1,276 @@
+        )
+
+    if check_derivs("D_geodesic", R_transform, Z_transform, L_transform):
+        data["D_geodesic"] = (
+            (
+                surface_integrals(
+                    grid,
+                    dS * mu_0 * dot(data["J"], data["B"]) / data["|grad(psi)|"] ** 3,
+                )
+            )
+            ** 2
+            - surface_integrals(grid, dS * data["|B|^2"] / data["|grad(psi)|"] ** 3)
+            * surface_integrals(
+                grid,
+                dS
+                * (mu_0 * dot(data["J"], data["B"])) ** 2
+                / (data["|B|^2"] * data["|grad(psi)|"] ** 3),
+            )
+        ) / (2 * jnp.pi) ** 6
+
+    if check_derivs("D_Mercier", R_transform, Z_transform, L_transform):
+        data["D_Mercier"] = (
+            data["D_shear"] + data["D_current"] + data["D_well"] + data["D_geodesic"]
+        )
+
+    return data
+
+
+def compute_magnetic_well(
+    R_lmn,
+    Z_lmn,
+    L_lmn,
+    p_l,
+    i_l,
+    Psi,
+    R_transform,
+    Z_transform,
+    L_transform,
+    pressure,
+    iota,
+    data=None,
+):
+    """Compute the magnetic well proxy for MHD stability.
+
+    Notes
+    -----
+        Implements equation 3.2 in M. Landreman & R. Jorge (2020)
+        doi:10.1017/S002237782000121X.
+
+    Parameters
+    ----------
+    R_lmn : ndarray
+        Spectral coefficients of R(rho,theta,zeta) -- flux surface R coordinate (m).
+    Z_lmn : ndarray
+        Spectral coefficients of Z(rho,theta,zeta) -- flux surface Z coordinate (m).
+    L_lmn : ndarray
+        Spectral coefficients of lambda(rho,theta,zeta) -- poloidal stream function.
+    p_l : ndarray
+        Spectral coefficients of p(rho) -- pressure profile.
+    i_l : ndarray
+        Spectral coefficients of iota(rho) -- rotational transform profile.
+    Psi : float
+        Total toroidal magnetic flux within the last closed flux surface (Wb).
+    R_transform : Transform
+        Transforms R_lmn coefficients to real space.
+    Z_transform : Transform
+        Transforms Z_lmn coefficients to real space.
+    L_transform : Transform
+        Transforms L_lmn coefficients to real space.
+    pressure : Profile
+        Transforms p_l coefficients to real space.
+    iota : Profile
+        Transforms i_l coefficients to real space.
+
+    Returns
+    -------
+    data : dict
+        Dictionary of ndarray, shape(num_nodes,) of the magnetic well parameter.
+
+    """
+    grid = R_transform.grid
+    data = compute_pressure(p_l, pressure, data=data)
+    data = compute_geometry(R_lmn, Z_lmn, R_transform, Z_transform, data=data)
+    data = compute_magnetic_field_magnitude(
+        R_lmn,
+        Z_lmn,
+        L_lmn,
+        i_l,
+        Psi,
+        R_transform,
+        Z_transform,
+        L_transform,
+        iota,
+        data=data,
+    )
+
+    if check_derivs("magnetic well", R_transform, Z_transform, L_transform):
```

Comment:
The changes here introduce a bug. It's easy to miss because the papers don't mention it, but it is important that the sign of all the jacobeans always be positive. I can revert this change later.

Reply:
Sorry about that! Good catch. It should be fixed now